subiquity/server/server.py

```diff
@@ -529,13 +532,13 @@ async def wait_for_cloudinit(self):
             self.cloud_init_ok = True
         log.debug("waited %ss for cloud-init", time.time() - ci_start)
         if "status: done" in status_txt:
-            log.debug("loading cloud config")
-            init = stages.Init()
-            init.read_cfg()
-            init.fetch(existing="trust")
-            self.cloud = init.cloudify()
-            if 'autoinstall' in self.cloud.cfg:
-                cfg = self.cloud.cfg['autoinstall']
+            log.debug(
+                "loading cloud config from %s", cloud_init_instance_data_path
+            )
+            with open(cloud_init_instance_data_path) as stream:
```

Comment:
In what scenarios can we see FileNotFoundError due to the instance data not being there?

Reply:
`/run/cloud-init/instance-data*json` will only not exist in the following scenarios:

 1. cloud-init is disabled: by `/etc/cloud/cloud-init.disabled`
 2. kernel cmdline param `cloud-init=disabled` exists 
 3. cloud-init being unable to detect a valid datasource on this current system boot so cloud-init's systemd generator `/lib/systemd/system-generators/cloud-init-generator` (which calls ds-identify)
 4. Some broad bug in cloud-init that breaks initial datasource detection (which we'd need to hotfix)

Generally in that disabled case subiquity should have fallen through the other logical path where it [assumes cloud-init is disabled](https://github.com/canonical/subiquity/blob/main/subiquity/server/server.py#L541-L544)  TBH, once we have JSON/YAML output format from `cloud-init status` I'd want to make the status output parsing a bit more strict than it currently is in server.py.

Reply:
Also once we add the ability to disable Desktop installer after `installed boot` then we know that the final booted desktop installed image will no longer surface /run/cloud-init/* files as cloud-init will be in disabled state.

Reply:
OK, for the check if cloud-init is disabled or not, from the Subiquity perspective, the planned machine-readable output will help with that.  So for now I'm prepared to assume that's correct.